source/multigrid/mg_transfer_prebuilt.cc

```diff
@@ -284,23 +284,41 @@ void MGTransferPrebuilt<VECTOR>::build_matrices (
   // We keep track in the bitfield dof_touched which global dof has
   // been processed already (on the current level). This is the same as
   // the multigrid running in serial.
-  // Only entering on the finest level gives wrong results (why?)
+
+  struct dof_pair
+  {
+    unsigned int level;
+    unsigned int global_dof_index;
+    unsigned int level_dof_index;
```

Comment:
I guess you want to make `global_dof_index` and `level_dof_index` of type `types::global_dof_index`